frontend/packages/dev-console/src/components/pipelines/pipeline-topology/factories.ts

```diff
@@ -39,29 +38,23 @@ export const componentFactory: ComponentFactory = (kind: ModelKind, type: string
   }
 };
 
-// TODO: Fix this hack as it's not the best way to get the layout update
-type CallbackLayout = (onLayout: LayoutCallback) => LayoutFactory;
-export const layoutFactory: CallbackLayout = (onLayout) => (type: string, graph: Graph) => {
+export const layoutFactory: LayoutFactory = (type: string, graph: Graph) => {
   switch (type) {
     case PipelineLayout.DAGRE_BUILDER:
     case PipelineLayout.DAGRE_VIEWER:
-      return new DagreLayout(
-        graph,
-        {
-          // Hack to get around undesirable defaults
-          // TODO: fix this, it's not ideal but it works for now
-          linkDistance: 20,
-          nodeDistance: 20,
-          groupDistance: 0,
-          collideDistance: 0,
-          simulationSpeed: 0,
-          chargeStrength: 0,
-          allowDrag: false,
-          layoutOnDrag: false,
-          ...getLayoutData(type),
-        },
-        onLayout,
-      );
+      return new DagreLayout(graph, {
+        // Hack to get around undesirable defaults
+        // TODO: fix this, it's not ideal but it works for now
+        linkDistance: 0,
+        nodeDistance: 0,
+        groupDistance: 0,
+        collideDistance: 0,
+        simulationSpeed: 0,
+        chargeStrength: 0,
+        allowDrag: false,
+        layoutOnDrag: false,
+        ...getLayoutData(type),
+      });
```

Comment:
Would it be better to override the `BaseLayout` defaults in `DagreLayout` or are the defaults undesirable specifically for this implementation?

Reply:
I guess that's the crux of the issue... What I did here was akin to a caveman stopping an alarm clock... I just hit it until it stopped doing the thing :)

I see the Dagre Layout implementation as a means to layout nodes in a specific "pattern". In my case, I need a very specific pattern that plays by the Dagre ruleset; mainly because I coded the whole thing with that in mind and have yet to determine an alternative way to do it after seeing your changes.

I suppose in the world of our layouts, we need to decide how we are going to approach them. I don't know much about the other layouts we have done, so I'm unsure of what we should / could do in this case. Ideally there are some commonalities we should be aware of.

`allowDrag`, `layoutOnDrag` for instance are important for a layout I would think. Either it makes it once and is finished, or does it update as the layout moves.

Things like `linkDistance` and `nodeDistance` sorta conflict with the way Dagre options are setup... and if I zero out my Dagre options, I still get the Base Layout adjustments.

I guess we could consider "grouping" of options... A layout library called VisJs took an interesting multi-type approach to their options: https://visjs.github.io/vis-network/docs/network/layout.html. Where they allowed a boolean OR an object that specified granular control. We could take this approach?

Reply:
OK, we can leave it be for now and revisit if/when we publicize the topology package or need another usage within the console.